pkg/operator/eventwatch/controller.go

```diff
@@ -47,13 +47,13 @@ func New() *Builder {
 	return &Builder{}
 }
 
-// WithEventHandler add handler for event matching the namespace and the reason.
+// WithEventHandler add handler for event matching the namespace and the reason substring.
 // This can be called multiple times.
-func (b *Builder) WithEventHandler(namespace, reason string, processEvent func(event *corev1.Event) error) *Builder {
+func (b *Builder) WithEventHandler(namespace, reasonSubstring string, processEvent func(event *corev1.Event) error) *Builder {
```

Comment:
I wonder whether this is really a good approach. I don't like this fuzzy matching. Alternatives: `*Panic` for postfix match.

Reply:
i was thinking to add some matcher (we use some in other places), but I thought it might be an overkill